tests/system/runtimes/test_nuclio.py

```diff
@@ -58,6 +59,29 @@ def test_deploy_function_with_error_handler(self):
 
         assert deployment == function.get_url()  # check function url
 
+    def test_deploy_function_with_model_runner(self):
+        code_path = str(self.assets_path / "function_with_model.py")
+
+        self._logger.debug("Creating nuclio function")
+        function = mlrun.code_to_function(
+            name="function_with_model",
+            kind="serving",
+            project=self.project_name,
+            filename=code_path,
+            image="mlrun/mlrun",
+        )
+
+        graph = function.set_topology("flow", engine="async")
+        model_runner_step = ModelRunnerStep()
+        model_runner_step.add_model("DummyModel", name="my-model")
+
+        graph.to(model_runner_step)
+
+        self._logger.debug("Deploying nuclio function")
+        deployment = function.deploy()
+
+        assert deployment == function.get_url()  # check function url
```

Comment:
please add invocaion, to see the full flow is working and also it is the only referance for now we have for understanding the new mechnisem.